### PR TITLE
Remove unnecessary functions in PrometheusRemoteWriteExporter

### PR DESCRIPTION
**Pull request by sincejune**
**Description:**
Modify code as suggested here: https://github.com/open-telemetry/opentelemetry-collector/pull/3184#discussion_r632754555

**Testing:**
All unit tests passed.

### CODE REVIEW

File: exporter/prometheusremotewriteexporter/exporter.go [hunk 1]
```diff
@@ -134,20 +134,68 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
...
+						}
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
+					case pdata.MetricDataTypeDoubleSum:
+						dataPoints := metric.DoubleSum().DataPoints()
+						if dataPoints.Len() == 0 {
 							dropped++
-							errs = append(errs, consumererror.Permanent(err))
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/exporter.go:159]: In a separate PR we should discuss about this, I don't think this should be an error.

File: exporter/prometheusremotewriteexporter/exporter.go [hunk 2]
```diff
@@ -134,20 +134,68 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
...
+						}
+					case pdata.MetricDataTypeIntGauge:
+						dataPoints := metric.IntGauge().DataPoints()
+						if dataPoints.Len() == 0 {
+							dropped++
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
+						}
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/exporter.go:154]: I think we can reduce some of the duplicate code here by having a func that accepts IntDataPointSlice. So this and IntSum will call into the same func, as well as for double.

```
func addIntDataPoints(dataPoints pdata.IntDataPointSlice, ...) error {
	if dataPoints.Len() == 0 {
		dropped++							
		return consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name()))
	}
	for x := 0; x &lt; dataPoints.Len(); x++ {	
		addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
	}
	return nil
}
```

Comment (sincejune) [reply exporter/prometheusremotewriteexporter/exporter.go]: Actually `addSingleIntDataPoint` requires the metric as a parameter. So I both passed datapoints and metric to the new function.

Comment (sincejune) [reply exporter/prometheusremotewriteexporter/exporter.go]: Updated in https://github.com/open-telemetry/opentelemetry-collector/pull/3261/commits/dcbc7ffc21dfc85e58b9ecdf4485106579a6d675

**Review by dashpole — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**